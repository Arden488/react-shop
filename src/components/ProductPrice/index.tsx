import * as React from 'react';

const ProductPrice = () => (
  <div>ProductPrice</div>
);

export default ProductPrice;
