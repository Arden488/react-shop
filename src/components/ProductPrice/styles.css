.product-price {

}

.product-price__main-value {
  position: relative;
  font-size: 34px;
}

.product-price__old-value {
  margin-left: 20px;
  font-size: 25px;
  color: #ccc;
  position: relative;
}

.product-price__discount {
  position: absolute;
  right: -20px;
  top: 0;
  font-size: 12px;
  background: red;
  color: #fff;
  display: block;
  height: 20px;
  width: 30px;
  text-align: center;
  line-height: 20px;
  border-radius: 4px;
}

.product-price__old-value::after {
  content: '';
  display: block;
  position: absolute;
  top: 50%;
  margin-top: -1px;
  height: 2px;
  width: 110%;
  left: -5%;
  background: rgb(139, 139, 139);
}
