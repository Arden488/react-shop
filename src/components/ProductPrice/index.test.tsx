import { shallow } from 'enzyme';
import * as React from 'react';

import ProductPrice from './index';

let component: React.ReactNode = null;

beforeEach(() => {
  component = shallow(<ProductPrice />);
});

it('renders correctly', () => {
  expect(component).toMatchSnapshot();
});
