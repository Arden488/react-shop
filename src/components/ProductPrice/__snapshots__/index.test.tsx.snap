// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="product-price"
>
  <span
    className="product-price__main-value"
  >
    $520
    <span
      className="product-price__discount"
    >
      -20%
    </span>
  </span>
  <span
    className="product-price__old-value"
  >
    $650
  </span>
</div>
`;
